File: hw10/HW10.java
package hw10;

public final class HW10 {
    private HW10() {
        //
    }

    public static int sumExponential(int n) {
        for (int base = 3; base < Math.ceil(Math.sqrt(n)); base++) {
            int nTemp = n;
            if (nTemp % base == 0) {
                while (nTemp > 0) {
                    int exp = (int)(Math.log(nTemp) / Math.log(base));
                    nTemp -= Math.pow(base, exp);
                    if (nTemp > Math.pow(base, exp)) break;
                }
                if (nTemp == 0) return base;
            }
        }
        return -1;
    }
}

File: hw10/hw10.js
function sumExponential(n) {
    for (let base = 3; base < Math.ceil(Math.sqrt(n)); base++) {
        let nTemp = n;
        if (nTemp % base === 0) {
            while (nTemp > 0) {
                let exp = Math.floor(Math.log(nTemp)/Math.log(base));
                nTemp -= base**exp;
                if (nTemp > base**exp) break;
            }
            if (nTemp === 0) return base;
        }
    }
    return -1;
}
